manticore/ethereum.py

```diff
@@ -448,6 +443,17 @@ def __str__(self):
     def address(self):
         return self._address
 
+    def _init_hashes(self):
+        #initializes self._hashes lazy
+        if self._hashes is None and self._seth is not None:
+            self._hashes = {}
+            md = self._seth.get_metadata(self._address)
+            if md is not None:
+                for signature, func_id in md.hashes.items():
+                    func_name = str(signature.split('(')[0])
+                    self._hashes[func_name] = signature, func_id
+            self._init_hashes = types.MethodType(lambda *args: None, self)
```

Comment:
is this line necessary? this feels pretty hacky to me, and i think we could omit it?

Reply:
It's faster if we keep it and slower if we remove it.
I think there is an alternative without lambda/Methodtype. Maybe you prefer this: 
 
```
def _null_func():
    pass

self._init_hashes = self._null_func 
```
 ??

Reply:
is it a micro-optimization? it's a tradeoff between speed and simplicity/readability of the code. I think we should strive towards a cleaner/simpler codebase unless the optimization _really_ matters.

just wanted to comment though, i'll leave this to your judgement! :)

Reply:
The most common complaint is about speed.
It's a pretty simple optimization that also captures the idea of it being a single execution.
I'll add some comment to improve readability and keep it.